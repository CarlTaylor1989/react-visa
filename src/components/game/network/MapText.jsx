import React from 'react';
import PropTypes from 'prop-types';

import './MapText.scss';

const MapText = (props) => {
  const { nameCoord, type } = props;
  const styles = {
    left: nameCoord.coordinates[0],
    top: nameCoord.coordinates[1]
  };
  return (
    <span
      dangerouslySetInnerHTML={{ __html: nameCoord.name }} // eslint-disable-line react/no-danger
      style={styles}
      className={`mapText ${type}`}
    />
  );
};

MapText.propTypes = {
  nameCoord: PropTypes.object.isRequired,
  type: PropTypes.string.isRequired
};

export default MapText;
