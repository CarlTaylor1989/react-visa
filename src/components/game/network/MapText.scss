@import '../../../styles/variables';

.mapText {
  &.familyName,
  &.productName {
    color: $colour-research;
    position: absolute;
    text-align: left;
    z-index: 1;
  }

  &.productName {
    color: $colour-white;
    font-size: 14px;
    font-weight: 600;
    line-height: 16px;
    max-width: 160px;

    &:not(.large) {
      &.small {
        max-width: 120px;
      }
    }

    &.smaller {
      max-width: 60px;
    }

    &.large {
      font-size: 36px;
      line-height: 40px;
      max-width: 400px;
      text-align: center;
      width: 400px;

      &.medium {
        width: 300px;
      }

      &.small {
        width: 250px;
      }

      &.smaller {
        width: 170px;
      }

      &.left {
        text-align: left;
      }
    }
  }

  &.familyName {
    font-size: 16px;
    font-weight: 800;

    &:not(.small) {
      white-space: nowrap;
    }

    &.small {
      width: 250px;
    }

    &.type1 {
      color: $colour-family-1;
    }

    &.type2 {
      color: $colour-family-2;
    }

    &.type3 {
      color: $colour-family-3;
    }

    &.type4 {
      color: $colour-family-4;
    }

    &.type5 {
      color: $colour-family-5;
    }
  }
}
