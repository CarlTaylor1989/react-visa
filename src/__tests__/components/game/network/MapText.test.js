/* global toJson */
import React from 'react'; // eslint-disable-line no-unused-vars
import MapText from '../../../../components/game/network/MapText';

let mockProps;
beforeEach(() => {
  mockProps = {
    nameCoord: {
      name: 'irrelevant',
      coordinates: [100, 100]
    },
    type: 'type'
  };
});

test('renders the MapText component', () => {
  const wrapper = shallow(<MapText {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});
