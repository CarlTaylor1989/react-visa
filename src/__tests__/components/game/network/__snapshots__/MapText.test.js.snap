// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the MapText component 1`] = `
<span
  className="mapText type"
  dangerouslySetInnerHTML={
    Object {
      "__html": "irrelevant",
    }
  }
  style={
    Object {
      "left": 100,
      "top": 100,
    }
  }
/>
`;
